src/types/chat.ts

```diff
@@ -11,6 +11,7 @@ export type BaseChatMessage = {
   ts_utc: firebase.firestore.Timestamp;
   deleted?: boolean;
   threadId?: string;
+  isQuestion?: boolean;
```

Comment:
should we extend `BaseChatMessage` with `isQuestion` prop instead of `VenueChatMessage` type ?

Reply:
Probably not, as I believe it’s only relevant in venue chat, not private chat/etc.

Reply:
oops I confused the types. :see_no_evil: 
Currently it extends `BaseChatMessage`. and I suggest to extend `VenueChatMessage` instead.

Reply:
Ah! Then yes, this should be on VenueChatMessage i believe

Reply:
So thinking about this even more, and looking at the implementation in https://github.com/sparkletown/sparkle/pull/1343/files?file-filters%5B%5D=.ts#diff-149830263521dd5458cda6d3607035614de3c2d8c55a4784ac632891361601e7 we actually probably want to be creating a new type altogether.

Following the pattern there, something like:

```typescript
export type QuestionMessage = BaseChatMessage & {
  isQuestion: true;
};

export type ChatMessage = PrivateChatMessage | VenueChatMessage | PollMessage | QuestionMessage;

export const isQuestionMessage = (r: unknown): r is QuestionMessage =>
  typeof r === "object" && isTruthy(r) && r.hasOwnProperty("isQuestion") && r.isQuestion === true;
```

Reply:
Though this still feels kind of hacky.. instead of adding a boolean to our message types, we should be more flexible/future thinking and just give them a specific `type`.

Then we could have similar code to above, but where we check `type`. This would make our message API simpler in general, and is cleaner/more robust.

For now I suggest we just add it to the 'question' messages here, and to the poll messages in that PR. It'll be way easier to implement now before they get shipped out than to have to deal with backfilling/handling legacy data across all of our environments in some future point once they hit production.

@mike-lvov Thoughts? I feel fairly strongly about this one.

Reply:
https://github.com/sparkletown/sparkle/pull/1424/commits/e9a0a74ef91a0b46ffa82e79da66cce4b306f740